pkg/istiovalues/fips.go

```diff
@@ -0,0 +1,46 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istiovalues
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/istio-ecosystem/sail-operator/pkg/helm"
+)
+
+const (
+	FipsEnableFilePath = "/proc/sys/crypto/fips_enabled"
+	TestModeFilePath   = "/tmp/fips_enabled"
+)
+
+// ApplyFipsValues: Detect FIPS enabled mode on OpenShift and set value pilot.env.COMPLIANCE_POLICY
+// To avoid changing the host machine FIPS_ENABLED_FILE_PATH configuration,
+// set testMode to true when running unit tests.
+func ApplyFipsValues(mergedHelmValues helm.Values, testMode bool) (helm.Values, error) {
+	fipsEnabledFile := FipsEnableFilePath
+	if testMode {
+		fipsEnabledFile = TestModeFilePath
+	}
+
+	contents, err := os.ReadFile(fipsEnabledFile)
+	if err == nil && string(contents) == "1\n" {
```

Comment:
Perhaps it's safer to trim the contents and compare them to `"1"` instead of relying on the `\n` always being there.